File: src/main/java/site/gachontable/gachontablebe/global/jwt/JwtAuthenticationEntryPoint.java
package site.gachontable.gachontablebe.global.jwt;

import com.fasterxml.jackson.databind.ObjectMapper;
import jakarta.servlet.http.HttpServletRequest;
import jakarta.servlet.http.HttpServletResponse;
import org.springframework.security.core.AuthenticationException;
import org.springframework.security.web.AuthenticationEntryPoint;
import org.springframework.stereotype.Component;
import site.gachontable.gachontablebe.global.error.ErrorResponse;

import java.io.IOException;

import static site.gachontable.gachontablebe.global.error.ErrorCode.EMPTY_AUTHENTICATION;

@Component
public class JwtAuthenticationEntryPoint implements AuthenticationEntryPoint {

    @Override
    public void commence(HttpServletRequest request, HttpServletResponse response,
                         AuthenticationException authException) throws IOException {
        setResponse(response);
    }

    public void setResponse(HttpServletResponse response) throws IOException {
        response.setContentType("application/json;charset=UTF-8");
        response.setStatus(HttpServletResponse.SC_UNAUTHORIZED);

        ObjectMapper mapper = new ObjectMapper();
        String jsonResponse = mapper.writeValueAsString(
                new ErrorResponse(EMPTY_AUTHENTICATION));

        response.getWriter().write(jsonResponse);
    }
}

File: src/main/kotlin/site/gachontable/infra/security/jwt/filter/JwtAuthenticationEntryPoint.kt
package site.gachontable.infra.security.jwt.filter

import com.fasterxml.jackson.databind.ObjectMapper
import jakarta.servlet.http.HttpServletRequest
import jakarta.servlet.http.HttpServletResponse
import org.springframework.security.core.AuthenticationException
import org.springframework.security.web.AuthenticationEntryPoint
import org.springframework.stereotype.Component
import site.gachontable.independent.error.ErrorResponse
import site.gachontable.independent.type.ErrorCode

@Component
class JwtAuthenticationEntryPoint : AuthenticationEntryPoint {
    override fun commence(
        request: HttpServletRequest,
        response: HttpServletResponse,
        authException: AuthenticationException,
    ) {
        setResponse(response)
    }

    fun setResponse(response: HttpServletResponse) {
        response.apply {
            contentType = "application/json;charset=UTF-8"
            status = HttpServletResponse.SC_UNAUTHORIZED
        }

        val mapper = ObjectMapper()
        val jsonResponse = mapper.writeValueAsString(
            ErrorResponse(ErrorCode.EMPTY_AUTHENTICATION)
        )

        response.writer.write(jsonResponse)
    }
}
